internal/datastore/post.go

```diff
@@ -70,6 +71,90 @@ func (d *delphisDB) GetPostsByDiscussionIDIter(ctx context.Context, discussionID
 	}
 }
 
+/* Equivalent of GetPostsByDiscussionIDIter, but accepting a cursor and a limit for fetching. In our implementation,
+   the cursor indicates the creation timestamp of the posts, allowing to fetch contents up to a certain date and time. */
+func (d *delphisDB) GetPostsByDiscussionIDFromCursorIter(ctx context.Context, discussionID string, cursor string, limit int) PostIter {
+	logrus.Debug("GetPostsByDiscussionIDFromCursorIter::SQL Query")
+	if err := d.initializeStatements(ctx); err != nil {
+		logrus.WithError(err).Error("GetPostsByDiscussionIDFromCursorIter::failed to initialize statements")
+		return &postIter{err: err}
+	}
+
+	rows, err := d.prepStmts.getPostsByDiscussionIDFromCursorStmt.QueryContext(
+		ctx,
+		discussionID,
+		cursor,
+		limit,
+	)
+	if err != nil {
+		logrus.WithError(err).Error("failed to query GetPostsByDiscussionIDFromCursorIter")
+		return &postIter{err: err}
+	}
+
+	return &postIter{
+		ctx:  ctx,
+		rows: rows,
+	}
+}
+
+func (d *delphisDB) GetPostsConnectionByDiscussionID(ctx context.Context, discussionID string, cursor string, limit int) (*model.PostsConnection, error) {
+	logrus.Debug("GetPostsConnectionByDiscussionID::SQL Query")
+	if err := d.initializeStatements(ctx); err != nil {
+		logrus.WithError(err).Error("GetPostsConnectionByDiscussionID::failed to initialize statements")
+		return nil, err
+	}
+
+	/* Note: An additional item is fetched beyond the requested limit. This is required
+	   to determine if at least one next page is present after the current one. */
+	iter := d.GetPostsByDiscussionIDFromCursorIter(ctx, discussionID, cursor, limit+1)
+	postArr, err := d.PostIterCollect(ctx, iter)
+	if err != nil {
+		logrus.WithError(err).Error("GetPostsConnectionByDiscussionID::failed to initialize statements")
+		return nil, err
+	}
+
+	edges := make([]*model.PostsEdge, 0)
+	for _, elem := range postArr {
```

Comment:
The produced PostsEdge list maintains the descending order imposed in the underlying SQL statament. Ordering elements inside a page by ascending order can be implemented easily if necessary, let me know if that is a requirement.  Due to how ListViews work in Flutter, it may make sense to reorder them in ascending order.